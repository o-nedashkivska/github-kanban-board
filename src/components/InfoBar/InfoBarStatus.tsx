import { Typography } from "antd";
import { BulbOutlined } from "@ant-design/icons";

import styles from "./info-bar.module.css";

const { Paragraph } = Typography;

const InfoBarStatus: React.FC = () => {
  return (
    <Paragraph className={styles["info-bar__status"]} italic>
      <BulbOutlined
        style={{ color: "#e67701" }}
        className={styles["info-bar__icon"]}
      />
      To view issues of any existing repository, please enter a valid repository
      url into the input above and click 'Load issues' button.
    </Paragraph>
  );
};

export default InfoBarStatus;
