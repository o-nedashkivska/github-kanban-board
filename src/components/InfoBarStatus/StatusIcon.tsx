import {
  BulbOutlined,
  WarningOutlined,
  LoadingOutlined,
} from "@ant-design/icons";

interface StatusIconProps extends React.HTMLProps<HTMLElement> {
  status?: string;
}

const StatusIcon: React.FC<StatusIconProps> = ({
  status = "initial",
  ...props
}) => {
  switch (status) {
    case "rejected":
      return <WarningOutlined style={{ color: "#ff4d4f" }} {...props} />;
    case "loading":
      return <LoadingOutlined style={{ color: "#52c41a" }} {...props} />;
    case "initial":
    default:
      return <BulbOutlined style={{ color: "#e67701" }} {...props} />;
  }
};

export default StatusIcon;
