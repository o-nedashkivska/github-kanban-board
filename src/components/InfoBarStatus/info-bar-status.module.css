.info-bar__status {
  padding-top: 10px;
}

.info-bar__icon {
  padding-right: 5px;
}
