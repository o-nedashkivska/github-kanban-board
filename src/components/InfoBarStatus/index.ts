import InfoBarStatus from "./InfoBarStatus";

export default InfoBarStatus;
