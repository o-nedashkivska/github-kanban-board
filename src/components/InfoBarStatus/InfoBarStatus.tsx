import { Typography } from "antd";
import { useSelector } from "react-redux";
import {
  getCurrentRepoStatus,
  getCurrentRepoStatusReason,
} from "../../store/selectors";
import StatusIcon from "./StatusIcon";

import styles from "./info-bar-status.module.css";

const { Paragraph } = Typography;

const InfoBarStatus: React.FC = () => {
  const status = useSelector(getCurrentRepoStatus);
  const statusReason = useSelector(getCurrentRepoStatusReason);

  return (
    <Paragraph className={styles["info-bar__status"]} italic>
      <StatusIcon status={status} className={styles["info-bar__icon"]} />
      {statusReason}
    </Paragraph>
  );
};

export default InfoBarStatus;
