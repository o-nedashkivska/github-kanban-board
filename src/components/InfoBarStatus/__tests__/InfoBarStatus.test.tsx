import { screen } from "@testing-library/react";

import { renderWithProviders } from "../../../test-utils";

import InfoBarStatus from "../InfoBarStatus";

describe("InfoBarStatus", () => {
  const preloadedState = {
    currentRepo: {
      name: "facebook/react",
      stars: 0,
      status: "rejected",
      statusReason: "Something went wrong.",
    },
  };

  test("displays status icon", () => {
    renderWithProviders(<InfoBarStatus />, { preloadedState });

    const statusIcon = screen.getByRole("img");

    expect(statusIcon).toBeInTheDocument();
  });

  test("displays correct status reason", () => {
    renderWithProviders(<InfoBarStatus />, { preloadedState });

    expect(
      screen.getByText(preloadedState.currentRepo.statusReason)
    ).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = renderWithProviders(<InfoBarStatus />, {
      preloadedState,
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
