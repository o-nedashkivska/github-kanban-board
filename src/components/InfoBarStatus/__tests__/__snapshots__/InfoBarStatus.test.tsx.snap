// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBarStatus matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-typography info-bar__status css-dev-only-do-not-override-1kuana8"
  >
    <i>
      <span
        aria-label="warning"
        class="anticon anticon-warning info-bar__icon"
        role="img"
        style="color: rgb(255, 77, 79);"
      >
        <svg
          aria-hidden="true"
          data-icon="warning"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M464 720a48 48 0 1096 0 48 48 0 10-96 0zm16-304v184c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V416c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8zm475.7 440l-416-720c-6.2-10.7-16.9-16-27.7-16s-21.6 5.3-27.7 16l-416 720C56 877.4 71.4 904 96 904h832c24.6 0 40-26.6 27.7-48zm-783.5-27.9L512 239.9l339.8 588.2H172.2z"
          />
        </svg>
      </span>
      Something went wrong.
    </i>
  </div>
</DocumentFragment>
`;
